mri/operators/linear/wavelet.py

```diff
@@ -52,28 +53,39 @@ def __init__(self, wavelet_name, nb_scale=4, verbose=0, dim=2,
         self.unflatten = unflatten
         self.n_jobs = n_jobs
         self.n_coils = n_coils
+        if self.n_coils == 1 and self.n_jobs != 1:
+            print("Making n_jobs = 1 for WaveletN as n_coils = 1")
+            self.n_jobs = 1
         self.backend = backend
         self.verbose = verbose
         if wavelet_name not in pysap.AVAILABLE_TRANSFORMS:
             raise ValueError(
                 "Unknown transformation '{0}'.".format(wavelet_name))
         transform_klass = pysap.load_transform(wavelet_name)
-        self.transform = transform_klass(
-            nb_scale=self.nb_scale, verbose=verbose, dim=dim, **kwargs)
+        self.transform_queue = []
+        n_proc = self.n_jobs
+        if n_proc < 0:
+            n_proc = joblib.cpu_count() + self.n_jobs + 1
+        # Create transform queue for parallel execution
+        for i in range(min(n_proc, self.n_coils)):
+            self.transform_queue.append(transform_klass(
+                nb_scale=self.nb_scale,
+                verbose=verbose,
+                dim=dim,
+                **kwargs)
+            )
         self.coeffs_shape = None
 
-    def get_coeff(self):
-        return self.transform.analysis_data
-
-    def set_coeff(self, coeffs):
```

Comment:
Where were these used?

Reply:
This was used earlier by user, but now we dont tend to use these. The idea back then in my opinion was to just set and get coefficients, which is now clearly done by op and adj_op anyway.

Reply:
ok cool